src/lib.rs

```diff
@@ -17,23 +17,28 @@
 //! [uProtocol Core API types](https://github.com/eclipse-uprotocol/up-core-api). The crate contains trait definitions and
 //! convenience functionality for building, converting, validating and serializing uProtocol data types.
 //!
-//! ## This crate includes:
+//! ## Library contents
 //!
-//! - the [cloudevents] module that offers a common way to represent uProtocol messages using the [cloudevents data model](https://cloudevents.io/)
-//! - the rpc module which offers wrappers for dealing with uProtocol payload in the context of RPC method invocation
-//! - the uattributes module with uProtocol message attribute types and validators
-//! - the umessage module which defines the uProtocol core message type and provides related convenience functionality
-//! - the upayload module which defines payload representation for uProtocol messages
-//! - the uri module providing convenience wrappers for creation and validation of uProtocol-style resource identifiers
-//! - the ustatus module which provices uProtocol types for representing status and status codes
-//! - the utransport module as an interface contract between uProtocol and specific transport protocol implementations
-//! - the uuid module which generates and validates UUIDs as per the uProtocol specification
+//! * `rpc` module, which offers wrappers for dealing with uProtocol payload in the context of RPC method invocation
+//! * `uattributes` module, with uProtocol message attribute types and validators
+//! * `umessage` module, which defines the uProtocol core message type and provides related convenience functionality
+//! * `upayload` module, which defines payload representation for uProtocol messages
+//! * `uri` module, providing convenience wrappers for creation and validation of uProtocol-style resource identifiers
+//! * `ustatus` module, which provices uProtocol types for representing status and status codes
+//! * `utransport` module, as an interface contract between uProtocol and specific transport protocol implementations
+//! * `uuid` module, which generates and validates UUIDs as per the uProtocol specification
 //!
 //! For user convenience, all of these modules export their types on up_rust top-level, except for (future) optional features.
 //!
+//! ## Optional features
+//!
+//! Some crate features are made optional, these include:
+//!
+//! * `cloudevents` feature, offering a way to represent uProtocol messages using the [cloudevents data model](https://cloudevents.io/)
+//!
```

Comment:
I'm not sure how "Rust-literate" we're expecting users to be. If we're expecting them to be fairly literate, maybe my comment not needed. Perhaps here we could provide a link to how to enable an optional feature in Rust or provide specifically how to enable that for this crate?

WDYT?

Reply:
Hm, maybe not in the crates.io description - but I'll add a bit of text to the project README, wdyt?